readthedocs/redirects/models.py

```diff
@@ -167,76 +234,96 @@ def get_full_path(
             filename=filename,
         )
 
-    def get_redirect_path(self, path, full_path=None, language=None, version_slug=None):
+    def get_redirect_path(self, filename, path=None, language=None, version_slug=None):
+        """
+        Resolve the redirect for the given filename.
+
+        .. note::
+
+           This method doesn't check if the current path matches ``from_url``,
+           that should be done before calling this method
+           using ``Redirect.objects.get_redirect_path_with_status``.
```

Comment:
The current name is `get_redirect_path`, and it's a method from the model (as opposite to a manager method). The note is more since this method used to do the matching too. I don't have an idea for another name, let me know if you think there is a more explicit name.

Reply:
I think it's fine for now. The risk is just an infinite redirect right, so not a security issue or anything, so fine to be less defensive I think.